src/chain.js

```diff
@@ -78,10 +83,22 @@ const chain = ({ request, source, head, handlerOpts, server, isPlain }) => {
 
         source.pipe(socket);
         socket.pipe(source);
+
+        // Once target socket closes forcibly, the source socket gets paused.
+        // We need to enable flowing, otherwise the socket would remain open indefinitely.
+        // Nothing would consume the data, we just want to close the socket.
```

Comment:
When this happens, it's also a good idea to set a timeout so we don't download the data indefinitely. Or limit the acceptable data to 2KB.

Reply:
Actually the data limit is not necessary since we have 16KB high watermark. I've added

```js
            if (sourceSocket.writable) {
                sourceSocket.end();
            }
```

so it gracefully closes the socket.